In [35]:
import pandas as pd

intakes = pd.read_csv("aac_intakes.csv/aac_intakes.csv")
intakes_and_outcomes = pd.read_csv("aac_intakes_outcomes.csv/aac_intakes_outcomes.csv")
outcomes = pd.read_csv("aac_outcomes.csv/aac_outcomes.csv")

Answers with decimals have been rounded to 2 decimal places.

## QUESTION 1

Is there an area where more pets are found?

The code below tells you where the 5 most common locations of found pets are, along with the number of pets found in their respective locations.

In [36]:
print(intakes.found_location.value_counts().head(6))

found_location
Austin (TX)                          14443
Outside Jurisdiction                   948
Travis (TX)                            921
7201 Levander Loop in Austin (TX)      517
Del Valle (TX)                         411
Pflugerville (TX)                      372
Name: count, dtype: int64


Answer: 
Yes, the top 5 places where the most pets were found were, list in descending order of pets found from highest to lowest, outside jurisdiction (948), Travis (TX) (921), 7201 Levander Loop in Austin (TX) (517), Del Valle (TX) (411) and Pflugerville (TX) (372).

## QUESTION 2

What is the average number of pets found in a month in the year 2015?

This is how I calculated my working, in which the code tells you the mean number of pets found in a month.

In [37]:
print(intakes_and_outcomes[intakes_and_outcomes['intake_year'] == 2015].intake_number.mean())

1.1407561901705974


Answer: The average number of pets found in a month in the year 2015 was 1 pet (1.14).

## QUESTION 3

Are there months where there is a higher number of animals found?

The code below lists the top 5 most common months where the most pets are found, along with the number of pets found for the respective months.

In [38]:
print(intakes_and_outcomes.intake_month.value_counts().head())

intake_month
5     7966
10    7652
6     7459
3     6613
11    6597
Name: count, dtype: int64


Answer: Yes, the top 5 months where the highest number of animals are found are, in descending order from highest to lowest, May (7966), October (7652), June (7459), March (6613) and November (6597).

## QUESTION 4

What is the ratio of incoming pets vs. adopted pets?

Please find my working below, in which the number of intakes and adopted pets are printed out respectively.

In [56]:
print(len(intakes))
print(len(outcomes[outcomes['outcome_type']=='Adoption']))

80187
34232


Answer: 

Ratio = Number of intakes / Adopted pets = 80187/34232 = 2.34

For every 2 intakes, 1 pet is adopted.

## QUESTION 5

What is the distribution of the types of animals in the shelter?

The code below lists the different animal types, along with their respective numbers.

In [75]:
print(intakes.animal_type.value_counts())

animal_type
Dog          45743
Cat          29659
Other         4434
Bird           342
Livestock        9
Name: count, dtype: int64


Answer: The shelter demographic consists of 45743 dogs, 29659 cats, 342 birds and 9 livestock, while 4434 of the pets are of different types of animals.

## QUESTION 6

What are the adoption rates for specific breeds?

Answer:

Below is the code which shows the 5 most adopted pets.

In [76]:
print(outcomes[outcomes['outcome_type']=='Adoption'].breed.value_counts().head())

breed
Domestic Shorthair Mix      10129
Labrador Retriever Mix       2404
Pit Bull Mix                 2382
Chihuahua Shorthair Mix      2293
Domestic Medium Hair Mix     1067
Name: count, dtype: int64


The top 5 most adopted pets are, in order of breed from most adopted to least adopted are Domestic Shorthair Mixes (10129), Labrador Retriever Mixes (2404), Pit Bull Mixes (2382), Chihuahua Shorthair Mixes (2293) and Domestic Medium Hair Mixes (1067).

Number of Domestic Shorthair Mixes:

In [77]:
print(len(outcomes[outcomes['breed']=='Domestic Shorthair Mix']))

23821


The number of Domestic Shorthair Mixes is 23821. Hence the adoption rate is: 10129/23821 = 42.52%.
The adoption rate for Domestic Shorthair Mixes is 42.52%.

Number of Labrador Retriever Mixes:

In [78]:
print(len(outcomes[outcomes['breed']=='Labrador Retriever Mix']))

4849


The number of Labrador Retriever Mixes is 4849. Hence the adoption rate is 2404/4849 = 49.58%.
The adoption rate for Labrador Retriever Mixes is 49.58%.

Number of Pit Bull Mixes:

In [79]:
print(len(outcomes[outcomes['breed']=='Pit Bull Mix']))

6361


The number of Pit Bull Mixes is 6361. Hence the adoption rate is 2382/6361 = 37.45%.
The adoption rate for Pit Bull Mixes is 37.45%.

Number of Chihuahua Shorthair Mixes:

In [80]:
print(len(outcomes[outcomes['breed']=='Chihuahua Shorthair Mix']))

4874


The number of Chihuahua Shorthair Mixes is 4874. Hence the adoption rate is 2293/4874 = 47.05%.
The adoption rate for Chihuahua Shorthair Mixes is 47.05%.

Number of Domestic Medium Hair Mixes:

In [81]:
print(len(outcomes[outcomes['breed']=='Domestic Medium Hair Mix']))

2368


The number of Domestic Medium Hair Mixes is 2368. Hence the adoption rate is 1067/2368 =  45.06%.
The adoption rate for Domestic Medium Hair Mixes is 45.06%.

## QUESTION 7
What are the adoption rates for different colorings?

Answer:

The top 5 coloring of animals:

In [82]:
print(outcomes[outcomes['outcome_type']=='Adoption'].color.value_counts().head())

color
Black/White    3814
Black          2757
Brown Tabby    1914
Tan/White      1179
Brown/White    1135
Name: count, dtype: int64


The top 5 coloring of animals in the shelter are Black/White (3814), Black (2757), Brown Tabby (1914), Brown (1179), White (1135).

Number of animals colored Black/White:

In [83]:
print(len(outcomes[outcomes['color']=='Black/White']))

8396


The number of animals colored Black/White is 8396. Hence the adoption rate is 3814/8396 = 45.43%
The adoption rate for animals colored Black/White is 45.43%.

Number of animals colored Black:

In [84]:
print(len(outcomes[outcomes['color']=='Black']))

6780


The number of animals colored Black is 6780. Hence the adoption rate is 2757/6780 = 40.66%.
The adoption rate for animals colored Black is 40.66%.

Number of animals colored Brown Tabby:

In [85]:
print(len(outcomes[outcomes['color']=='Brown Tabby']))

4531


The number of animals colored Brown Tabby is 4531. Hence the adoption rate is 1914/4531 = 42.24%
The adoption rate for animals colored Brown Tabby is 42.24%.

Number of animals colored Brown:

In [86]:
print(len(outcomes[outcomes['color']=='Brown']))

3621


The number of animals colored Brown is 3621. Hence the adoption rate is 1179/3621 = 32.56%
The adoption rate for animals colored Brown is 32.56%.

Number of animals colored White:

In [87]:
print(len(outcomes[outcomes['color']=='White']))

2865


The number of animals colored White is 2865. Hence the adoption rate is 1135/2865 = 39.62%.
The adoption rate for animals colored White is 39.62%.

## QUESTION 8

About how many animals are spayed/neutered each month?

Below are the number of neutered male and spayed female animals respectively.

In [88]:
print(len(intakes_and_outcomes['sex_upon_outcome']=='Neutered Male'))
print(len(intakes_and_outcomes['sex_upon_outcome']=='Spayed Female'))

79672
79672


The number of animals that are spayed/neutered are 79672.

Number of animals spayed/neutered in a month = 79672/12 = 6414.33

Answer: 6414 animals